src/ctype/CTypeMetadata.ts

```diff
@@ -0,0 +1,17 @@
+import ICTypeMetadata, { IMetadata } from '../types/CTypeMetadata'
+import CType from './CType'
+import { WrapperMetadata } from './CTypeSchema'
+import * as CTypeUtils from './CTypeUtils'
+
+export default class CTypeMetadata implements ICTypeMetadata {
+  public ctypeHash: ICTypeMetadata['ctypeHash']
+  public metadata: ICTypeMetadata['metadata']
+
+  public constructor(ctype: CType['hash'], ctypeMetadata: IMetadata) {
```

Comment:
Why are using `IMetadata` here, when you use `ICTypeMetadata['metadata']` at other places?

Reply:
I guess it seemed like it should be only that. But, I understand it is more consistent.

Reply:
Also, this seems a little strange that I had `ICTypeMetadata['ctypeHash']` and `CType['hash']` should I just use the `ICTypeMetadata['ctypeHash']`?

Reply:
It's basically a reference to each other so it makes no difference, right?

Reply:
right

Reply:
we should just be consistent